packages/next/src/build/index.ts

```diff
@@ -2612,25 +2611,21 @@ export default async function build(
               // revalidate periods and dynamicParams settings
               appStaticPaths.forEach((routes, originalAppPath) => {
                 const encodedRoutes = appStaticPathsEncoded.get(originalAppPath)
-                const appConfig = appDefaultConfigs.get(originalAppPath) || {}
+                const appConfig = appDefaultConfigs.get(originalAppPath)
 
                 routes.forEach((route, routeIdx) => {
                   defaultMap[route] = {
                     page: originalAppPath,
                     query: { __nextSsgPath: encodedRoutes?.[routeIdx] },
-                    _isDynamicError: appConfig.dynamic === 'error',
+                    _isDynamicError: appConfig?.dynamic === 'error',
                     _isAppDir: true,
+                    _supportsPPR: appConfig
+                      ? isPPRSupported(config.experimental.ppr, appConfig)
+                      : undefined,
                   }
                 })
               })
 
-              // Ensure we don't generate explicit app prefetches while in PPR.
-              if (config.experimental.ppr && appPrefetchPaths.size > 0) {
```

Comment:
would we still want this invariant in the case where the `NextConfig` has a PPR matcher for this route?

Reply:
The `appPrefetchPaths` was a holdover from pre-PPR support for static prefetches. I've removed it as a part of this work, so the invariant here isn't required 🙌🏻